### PR TITLE
Smart mode for index page

### PR DESCRIPTION
**Pull request by cardil**
Shows HTML page when called from Web Browser. Shows JSON when called from command line or when application/json is explicitly listed in Accept header.

### CODE REVIEW

File: expressjs/src/routes/index.js [hunk 1]
```diff
@@ -24,6 +27,14 @@ module.exports = app => {
   })
 }
 
+function shouldRenderJson(req) {
+  const ua = req.header('user-agent')
+  if (ua && ua.startsWith('Mozilla')) {
```
Comment (rhuss) [inline expressjs/src/routes/index.js:32]: Are you sure that all browser's agent start with "Mozilla" ? I don't think so, I think we should be more selective and check that at least for Safari, Chrome, Firefox, MS Edge you get back a web page. Or there is even a better way to detect a browser.

Comment (cardil) [reply expressjs/src/routes/index.js:32]: All modern browsers have that. I did a check. See: https://www.whatismybrowser.com/guides/the-latest-user-agent/

**Pull request merged**